app/templates/search.html

```diff
@@ -15,23 +16,38 @@
         <div class="container">
             <form class="form-inline global-search">
                 <div class="form-group">
-                    <input class="searchbar" id="input" type="text" name="search" class="form-control" placeholder="Enter a game to get started" autofocus>
+                    {% macro searchbar(value) -%}
+                        <input class="searchbar" id="input" type="text" name="search" class="form-control" placeholder="Enter a game to get started" value="{{ value|e }}" autofocus>
+                    {%- endmacro %}
+                    {% if game is defined %}
+                        {{ searchbar(game.name) }}
+                    {% elif query is defined %}
+                        {{ searchbar(query) }}
+                    {% else %}
+                        {{ searchbar("") }}
+                    {% endif %}
                 </div>
                 <button type="submit" class="btn btn-info">Submit</button>
             </form>
-            {% if didyoumean is defined %}
-                 <h2 class="suggested">We couldn't find anything for <span class="emphasize">{{query}}</span>! Did you mean: <a href="/?search={{corrected_query}}"><span class="emphasize">{{didyoumean}}</span></a></h2>
+            {% if didyoumean1 is defined %}
+                 <h2 class="suggested">
+                    We couldn't find anything for <span class="emphasize">{{query|e}}</span>!
+                    Did you mean: <a href="/?app_id={{ didyoumean1.app_id }}"><span class="emphasize">{{ didyoumean1.name }}</span></a> or <a href="/?app_id={{ didyoumean2.app_id }}"><span class="emphasize">{{ didyoumean2.name }}</span></a>?
+                </h2>
+            {% endif %}
+            {% if no_such_app_id is defined %}
+                <h2 class="suggested">Sorry, but this app ID is not in our database! Don't hesitate to get in touch with us and will consider adding it.</h2>
             {% endif %}
         </div>
 
         {% if ranking is defined %}
             <!-- With love from https://github.com/BlackrockDigital/startbootstrap-creative -->
             <div class="container-fluid">
-                <h2 class="result-head">Showing results for: <span class="emphasize">{{query}}</span></h2>
+                <h2 class="result-head">Showing results for: <a href="{{ game.steam_url() }}"><span class="emphasize">{{ game.name }}</span></a></h2>
```

Comment:
@hunruh Thoughts on this jazz? Also, I can't seem to make that link clickable for the life of me